app/src/main/java/ru/prometeydev/movie/ui/MainActivity.kt

```diff
@@ -2,19 +2,41 @@ package ru.prometeydev.movie.ui
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.work.*
+import org.koin.android.ext.android.inject
 import ru.prometeydev.movie.R
 import ru.prometeydev.movie.common.setAsRoot
+import ru.prometeydev.movie.model.MoviesRepository
+import ru.prometeydev.movie.service.MoviesWorker
 import ru.prometeydev.movie.ui.movieslist.MoviesListFragment
+import java.util.concurrent.TimeUnit
 
 class MainActivity : AppCompatActivity() {
 
+    private val constraints = Constraints.Builder()
```

Comment:
Can you create some work manager starter component to remove implementation details from Activity?

Reply:
Yes, I can.